src/rabbit_sharding_shard.erl

```diff
@@ -93,14 +93,20 @@ bind_queues(#exchange{name = XName} = X) ->
 declare_queue(XName, Durable, N, Node) ->
     QBin = make_queue_name(exchange_bin(XName), a2b(Node), N),
     QueueName = rabbit_misc:r(v(XName), queue, QBin),
-    try rabbit_amqqueue:declare(QueueName, Durable, false, [], none,
-                                ?SHARDING_USER, Node) of
+    Args = [{<<"x-queue-master-locator">>, longstr, <<"client-local">>}],
+    try
+        rpc:call(Node, rabbit_amqqueue, declare, [QueueName, Durable, false, Args, none, ?SHARDING_USER, Node]) of
```

Comment:
Is the `rpc:call` necessary because the only way to specify the master node for a quorum queue is to do the declare on that node?

Reply:
Yes, we have to override any `queue-master-locator` policy and that's the only way to tell it in which exactly node we want it. But as @michaelklishin said below, it won't work on an upgraded system as it changes the queue arguments :(